pkg/unions/kubernetes_ingress.go

```diff
@@ -43,3 +44,64 @@ func NewKubernetesIngressFromParams(ingressParams generator.IngressParams) *Kube
 func (ki *KubernetesIngress) IsGenerated() bool {
 	return ki.isGenerated
 }
+
+func (ki *KubernetesIngress) GetName() string {
+	if ki.NetworkingV1Ingress != nil {
+		return ki.NetworkingV1Ingress.GetName()
+	} else if ki.ExtensionV1Beta1Ingress != nil {
+		return ki.ExtensionV1Beta1Ingress.GetName()
+	}
+	return ""
+}
+
+func (ki *KubernetesIngress) GetProtocol() string {
+	if (ki.NetworkingV1Ingress != nil && len(ki.NetworkingV1Ingress.Spec.TLS) > 0) || (ki.ExtensionV1Beta1Ingress != nil && len(ki.ExtensionV1Beta1Ingress.Spec.TLS) > 0) {
+		return "https"
+	}
+	return "http"
+}
+
+func (ki *KubernetesIngress) GetHost() string {
+	if ki.NetworkingV1Ingress != nil {
+		return ki.NetworkingV1Ingress.Spec.Rules[0].Host
+	} else if ki.ExtensionV1Beta1Ingress != nil {
+		return ki.ExtensionV1Beta1Ingress.Spec.Rules[0].Host
+	}
+	return ""
+}
+
+func (ki *KubernetesIngress) GetURLString() string {
+	return fmt.Sprintf("%v://%v", ki.GetProtocol(), ki.GetHost())
+}
+
+type KubernetesIngressList struct {
+	Items []*KubernetesIngress
+}
+
+func NewEmptyKubernetesIngressList() *KubernetesIngressList {
+	return &KubernetesIngressList{}
+}
+
+func (kil *KubernetesIngressList) GetNetworkingV1IngressList(skipIfExtensionV1Set bool) *v1.IngressList {
```

Comment:
Please add a comment describing the varaible `skipIfExtensionV1Set` in this and the function below

Reply:
Adding